.github/workflows/build.yml

```diff
@@ -0,0 +1,79 @@
+name: Build ml-metadata
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Upgrade pip
+      run: |
+        python -m pip install --upgrade pip pytest
+
+    - name: Build the package for Python ${{ matrix.python-version }}
+      run: |
+        version="${{ matrix.python-version }}"
+        DOCKER_SERVICE=manylinux-python$(echo "$version" | sed 's/\.//')
+        docker compose build ${DOCKER_SERVICE}
+        docker compose run ${DOCKER_SERVICE}
+
+    - name: Upload wheel artifact for Python ${{ matrix.python-version }}
+      uses: actions/upload-artifact@v4.4.0
+      with:
+        name: ml-metadata-wheel-py${{ matrix.python-version }}
+        path: dist/*.whl
+
+    - name: Install built wheel
+      run: pip install dist/*.whl
+
+    - name: Test
+      run: |
+        # cleanup (interferes with tests)
+        rm -rf bazel-*
+        # run tests
+        pytest -vv
```

Comment:
What are our (cc: @peytondmurray) thoughts on abstracting the build steps into a reusable/composite action and calling that from separate test and upload workflows? I'm not sure I'm really a fan of putting tests within the build/upload workflow.

Reply:
Yep, I'm a fan since we need the wheel to do pretty much anything that involves running ml-metadata code.

Reply:
[Here's an example of what I had in mind](https://github.com/smokestacklightnin/data-validation/tree/ci/testing/use-pytest/.github)

Reply:
Done.